library/network_connections.py

```diff
@@ -1493,8 +1495,7 @@ def check_activated(ac, dev):
             except AttributeError:
                 ac_reason = None
 
-            if dev:
-                dev_state = dev.get_state()
+            dev_state = dev.get_state() if dev else None
```

Comment:
I believe my previous comment was lost, if not, please excuse the duplicate.
CodeQL properly identifies this code as problematic. This change only makes CodeQL happy but does not really work towards refactoring the code to make it easier to understand. In the following lines `dev` and `dev_state` are  used without checking whether they are None and I guess it is also not needed since there is some implicit understanding whether this function will actually be called with `dev` not being a true value. I would rather see this function being properly refactored to make full use of CodeQL and creating additional benefit for the future.

Reply:
Thanks for the detailed explanation, @tyll ! Reverted.